integration-tests/test_register.py

```diff
@@ -0,0 +1,47 @@
+import contextlib
+import pytest
+import logging
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_register(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    response=insights_client.run('--register', check=False)
+    assert response.returncode == 0,\
+        "An application returns 0 after successful registration"
+    stdout = response.stdout.decode("UTF-8")
+    assert "Successfully registered host" in stdout,\
+        "An application should inform that a system was registered"
+    assert "Starting to collect Insights data" in stdout,\
+        "An application should inform that Insights data will be collected"
+    assert "Successfully uploaded report" in stdout,\
+        "An application should inform that report was successfully uploaded"
+    assert "View the Red Hat Insights console" in stdout,\
+        "An application should inform about Insights console"
+    assert insights_client.is_registered,\
+        "Current state of registration should be 'registered'"
+    
+def test_register_twice(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    insights_client.run('--register')
+    assert insights_client.is_registered
+    
+    response = insights_client.run('--register',check=False)
+    assert response.returncode == 0, "An application should return 0"
```

Comment:
ditto

Reply:
A way I wrote it is more descriptive and no hidden behavior is used. ie.  the test is rather a sort of documentation.